light-client/README.md

```diff
@@ -59,6 +59,87 @@ loop {
 }
 ```
 
+## Testing
+
+The Tendermint Light Client is primarily tested through unit tests.
+
+### Core Verification
+
+The logic for the core verification of light blocks is entirely self-contained in
+the [`predicates`](./src/predicates.rs) module. This code is exercised in a family
+of tests called `single_step` via a set of [JSON fixtures](./tests/support/single_step)
+which encode an initial trusted state, a target block to verify, and the
+expected result of the core verification algorithm.
+
+These tests come in two flavours:
+
+- `skipping` tests, where there is a gap between the initial trusted state and the target block.
+- `sequential` tests, where there the initial trusted state and the target block are adjacent.
+
+The following command can be used to run only these tests:
+
+```bash
+$ cargo test -p tendermint-light-client --test light_client single_step
+```
+
+### Bisection
+
+Similarly to the core verification logic, the algorithm for performing bisecting
+verification is exercised via a set of [JSON fixtures](./tests/support/bisection/single_peer)
+which encode an initial trusted state, a target block to verify, a set of intermediary blocks,
+and the expected result of the bisection algorithm.
+
+These tests target the [`light_client`](./src/light_client.rs) module,
+and can be found in the [`tests/light_client.rs`](./tests/light_client.rs) file.
+
+To run the tests:
+
+```bash
+$ cargo test -p tendermint-light-client --test light_client bisection
+```
+
+### Fork Detection
+
+Similarly to the bisection algorithm, the fork detection algorithm is tested via a set
+of [JSON fixtures](./tests/support/bisection/single_peer) which encode an initial trusted
+state, a target block to verify, a set of intermediary blocks, and the expected result
+of the fork detection algorithm.
+
+These tests target the [`supervisor`](./src/supervisor.rs) module,
+and can be found in the [`tests/supervisor.rs`](./tests/supervisor.rs) file.
+
+To run the tests:
+
+```bash
+$ cargo test -p tendermint-light-client --test supervisor
+```
+
+### Voting Power Calculator
+
+The [voting power calculator](./src/operations/voting_power.rs) is exercised through
+unit tests which rely on [JSON fixtures](./tests/support/voting_power/) to provide
+the calculator with various types of *light blocks* together with the expected result
+of the computation.
+
+The following command can be used to run only these tests:
+
+```bash
+$ cargo test -p tendermint-light-client voting_power
+```
+
+### Integration Tests
+
+This project also includes simple integration test which spawns a light client instance
+against a single Tendermint full node which acts both as a primary peer and as its
+own witness.
+
+Because this test requires a running Tendermint node, it is ignored by default.
+To run this test locally:
+
+```bash
+$ cargo test -p tendermint-light-client --test integration -- --ignored --nocapture
```

Comment:
Would it perhaps be a good idea here to be more explicit about the need to run a Tendermint node in order to run these integration tests?

```suggestion
# In one terminal
$ mkdir -p /tmp/tendermint
$ docker run -it --rm -v "/tmp/tendermint:/tendermint" tendermint/tendermint init
$ docker run -it --rm -v "/tmp/tendermint:/tendermint" -p 26657:26657 tendermint/tendermint node --proxy_app=kvstore

# In another terminal
$ cargo test -p tendermint-light-client --test integration -- --ignored --nocapture
```

Reply:
Good idea!